pup-0003.md

```diff
@@ -0,0 +1,154 @@
+barrier to entry for community members. With the cherry-pick model, all PRs are
+opened against master and the onus to update release branches is on the person
+merging (who usually has better knowledge of which branches a commit
+needs to be applied to).
+
+We've also seen some serious mistakes where branches are accidentally merged
+into other branches.  These situations have occurred multiple times, and each
+time required multiple people working several hours to resolve the problem.
+After being resolved the git history is convoluted with huge negative commits
+which later caused confusion at release time.  Commits could be accidentally
+cherry-picked but this would be less disastrous since it's only a single commit.
+
+Changes that we want to only be made to older release branches (but not to
+newer ones) still need to be merged forward. We could elminate the need for
+this with cherry-picking. It's worth noting though that this rarely happens.
+
+## Detailed Design
+
+This proposal would keep the current branching scheme[1] involving master,
+x.y-dev branches, and x.y-release branches. The workflow by which changes are
+moved between the branches would however change.
+
+The process works a bit differently depending on whether the change is a new
+feature, bug fix, or hotfix.
+
+#### New Feature
+
+1. All feature PRs are opened and merged against master.
+
+#### Bug Fix
+
+1. All bug fix PRs are opened against master
+2. After the PR is ready to be merged, the person merging the commit would
+   first merge the commit to master. Then, this person cherry-picks this change
+   from master to the latest x.y-dev branch.
+3. The person cherry-picking would also need to track in redmine which release
+   branches the change has been made to.
+
+#### Hotfix
+
+The current hotfix process[2] would remain basically unchanged. However,
+instead of merging forward the commit, we'd use cherry-picking.
+
+1. The affected release would be branched and a hotfix would be applied.
+2. After the release, the hotfix gets cherry-picked into the x.y-release branch.
+3. The change is then cherry-picked into the x.y-dev branch and master.
+4. The person cherry-picking would also need to track in redmine which release
+   branches the change has been made to.
+
+### Conflicts
+
+If the cherry-pick has conflicts, we should consider abandoning the cherry-pick
+and encourage users to upgrade to the next y release.
+
+### Using git cherry-pick
+
+When we perform the cherry-pick in git, we should use the `-x` flag which adds
+the original commit id to the commit message of the cherry-pick. This will help
+track changes across branches.
+
+### Updating Redmine
+
+When a cherry-pick is pushed to a release branch, the release field of the issue
+in redmine should be set to x.y.next. For example, when a cherry-pick is pushed
+to 2.13-dev, the issue's release field should be set to 2.13.next. Then when the
```

Comment:
In L#95 it says release branch, but I think of those as having `-release` in the name. So in L#95 is it s/release/dev/ ?